### PR TITLE
Add error message string to MoveItErrorCode msg

### PR DESCRIPTION
**Pull request by Abishalini**
When planning with MoveIt and MTC, we often assign the FAILURE error code. This is in an effort to add more description on why planning/execution failed.

### CODE REVIEW

File: msg/MoveItErrorCodes.msg [hunk 1]
```diff
@@ -40,3 +40,5 @@ int32 ABORT=-30
 
 # kinematics errors
 int32 NO_IK_SOLUTION=-31
+
+string error_message
```
Comment (sjahr) [inline msg/MoveItErrorCodes.msg:45]:
```suggestion
// Error message to provide additional information
string error_message
```

**Review by sjahr — COMMENTED** [review]
What do you think about adding an optional source string too?
E.g. like this:
```
// Name of the component that created the status
string source;
```

**Review by sjahr — APPROVED** [review]

File: msg/MoveItErrorCodes.msg [hunk 2]
```diff
@@ -40,3 +40,10 @@ int32 ABORT=-30
 
 # kinematics errors
 int32 NO_IK_SOLUTION=-31
+
+# A message to provide additional information.
+string error_message
```
Comment (sjahr) [inline msg/MoveItErrorCodes.msg:45]:
```suggestion
string error_message
```
```suggestion
string message
```

File: msg/MoveItErrorCodes.msg [hunk 3]
```diff
@@ -40,3 +40,10 @@ int32 ABORT=-30
...
+
+# Name of the component that created the status.
+# This is helpful to locate error source.
+string error_source
```
Comment (sjahr) [inline msg/MoveItErrorCodes.msg:49]:
```suggestion
string source
```

**Review by sjahr — COMMENTED** [review]
Nit: Maybe consider removing the `error_*` prefix to avoid code like this:
```
MoveItErrorCode error_code;
error.error_message = "Ahh";
```

**Review by adlarkin — APPROVED** [review]

**Review by sea-bass — COMMENTED** [review]
I think this is conflating a bunch of things.

It makes more sense to me to have a `MoveItError` message which contains the `MoveItErrorCode` code plus a string message?

**Comment by sjahr** [comment]
> I think this is conflating a bunch of things.
> 
> It makes more sense to me to have a `MoveItError` message which contains the `MoveItErrorCode` code plus a string message?

We'd need to do a lot of integration work by introducing an additional message type for that. What I like about having it in this message is that the message + the source information is immediately everywhere available where this message is used and we don't need to break anything

**Comment by simonschmeisser** [comment]
Please consider extending the 'enum' where necessary as well. Strings are hard to translate and error prone to handle in logic/error mitigation

**Comment by rhaschke** [comment]
> If this error message will be used as the "human-readable" description of the enum, that's very good and let's go ahead.

The message shouldn't serve as a description of the enum - this can be created statically and we already have a method to do so. Rather, the message should be used (if at all) to provide some more specific reason of the failure, e.g. why planning failed.

**Pull request merged**